kernel/include/ksched.h

```diff
@@ -46,7 +53,11 @@ static inline int _is_idle_thread(void *entry_point)
 
 static inline int _is_idle_thread_ptr(k_tid_t thread)
 {
+#ifdef CONFIG_SMP
+	return thread->base.is_idle;
```

Comment:
if we are checking idle thread ptr based on 'is_idle' then any thread which sets this filed becomes idle thread, do we want this ? idle thread is something which should be run at the system level and should be created early init right?

Reply:
Nothing sets this field at runtime, and threads do not "become idle".  An "idle thread" is a specific thread created at system startup.  There used to be one, so you would test it with "tid == _idle_thread".  Now there needs to be one per CPU so it needs a bit somewhere in the struct to tell us.